hoomd/hpmc/pytest/test_polyhedron.py

```diff
@@ -1,32 +1,210 @@
+    poly.shape['A'] = args_4
+    assert not poly.shape['A']['ignore_statistics']
+    for key in args_4.keys():
+        assert poly.shape['A'][key] == args_4[key]
+
+
+def test_poly_after_attaching(device,
+                              dummy_simulation_factory):
+
+    poly = hpmc.integrate.Polyhedron(2346)
+    poly.shape['A'] = args_4
+    poly.shape['B'] = args_5
+
+    sim = dummy_simulation_factory(particle_types=['A', 'B'])
+    sim.operations.add(poly)
+    sim.operations.schedule()
+
+    assert not poly.shape['A']['ignore_statistics']
+    assert poly.shape['B']['ignore_statistics']
+    for key in args_4.keys():
+        assert poly.shape['A'][key] == args_4[key]
+        assert poly.shape['B'][key] == args_5[key]
+
+
+def test_overlaps(device, lattice_simulation_factory):
+    mc = hoomd.hpmc.integrate.Polyhedron(23456, d=0, a=0)
+    mc.shape['A'] = dict(vertices=[(0, (0.75**0.5) / 2, -0.5),
+                                   (-0.5, -(0.75**0.5) / 2, -0.5),
+                                   (0.5, -(0.75**0.5) / 2, -0.5),
+                                   (0, 0, 0.5),
+                                   (0, 0, 0)],
+                         faces=[(0, 3, 1),
+                                (0, 2, 3),
+                                (1, 3, 2),
+                                (1, 2, 4),
+                                (0, 1, 4),
+                                (0, 4, 2)],
+                         overlap=[True, True, True, True, True, True])
+    sim = lattice_simulation_factory(dimensions=2, n=(2, 1), a=0.25)
+    sim.operations.add(mc)
+    sim.operations.schedule()
+    sim.run(1)
+    assert mc.overlaps > 0
+
+    s = sim.state.snapshot
+    if s.exists:
+        s.particles.position[0] = (0, 0, 0)
+        s.particles.position[1] = (0, 8, 0)
+    sim.state.snapshot = s
+    assert mc.overlaps == 0
+
+    s = sim.state.snapshot
+    if s.exists:
+        s.particles.position[0] = (0, 0, 0)
+        s.particles.position[1] = (0, 0.85, 0)
+    sim.state.snapshot = s
+    assert mc.overlaps > 0
+
+
+def test_shape_moves(device, lattice_simulation_factory):
+    mc = hoomd.hpmc.integrate.Polyhedron(23456)
+    mc.shape['A'] = dict(vertices=[(0, (0.75**0.5) / 2, -0.5),
+                                   (-0.5, -(0.75**0.5) / 2, -0.5),
+                                   (0.5, -(0.75**0.5) / 2, -0.5),
+                                   (0, 0, 0.5),
+                                   (0, 0, 0)],
+                         faces=[(3, 1, 2),
+                                (3, 0, 1),
+                                (3, 2, 0),
+                                (4, 2, 1),
+                                (4, 0, 2),
+                                (4, 1, 0)],
+                         overlap=[0, 0, 0, 0, 0, 0])
+    sim = lattice_simulation_factory()
+    sim.operations.add(mc)
+    sim.operations.schedule()
+    sim.run(100)
+    accepted_rejected_rot = sum(sim.operations.integrator.rotate_moves)
```

Comment:
It would be nice to check that `rotate_moves` and other loggable quantities properly return None when not attached.

Reply:
Added ec1b7eb0e